core/src/impl/Kokkos_SharedAlloc.hpp

```diff
@@ -146,10 +146,17 @@ class SharedAllocationRecord<void, void> {
  public:
   virtual std::string get_label() const { return std::string("Unmanaged"); }
 
+#if defined(__EDG__) && !defined(KOKKOS_COMPILER_INTEL)
```

Comment:
Why aren't you using `KOKKOS_COMPILER_NVHPC` instead? There is another compiler with the same problem?

Reply:
That's fair.
I was going for consistency with https://github.com/kokkos/kokkos/blob/2ad88ec38922556d4459c4dbd2813845f4b93356/core/src/impl/Kokkos_BitOps.hpp#L98-L101